addons/calendar/models/calendar.py

```diff
@@ -535,19 +535,33 @@ def todate(date):
             return val.astimezone(timezone)
 
         timezone = pytz.timezone(self._context.get('tz') or 'UTC')
-        startdate = pytz.UTC.localize(fields.Datetime.from_string(self.start))  # Add "+hh:mm" timezone
-        if not startdate:
-            startdate = datetime.now()
+        if both:
+            startdate = pytz.UTC.localize(fields.Datetime.from_string(self.start))  # Add "+hh:mm" timezone
+            stopdate = pytz.UTC.localize(fields.Datetime.from_string(self.stop))  # Add "+hh:mm" timezone
+            if not startdate:
+                startdate = datetime.now()
+            if not stopdate:
+                stopdate = startdate
+        else:
+            stopdate = False
 
         # Convert the start date to saved timezone (or context tz) as it'll
         # define the correct hour/day asked by the user to repeat for recurrence.
         startdate = startdate.astimezone(timezone)  # transform "+hh:mm" timezone
         rset1 = rrule.rrulestr(str(self.rrule), dtstart=startdate, forceset=True)
+        if both:
+            stopdate = stopdate.astimezone(timezone)  # transform "+hh:mm" timezone
+            rset2 = rrule.rrulestr(str(self.rrule), dtstart=stopdate, forceset=True)
         recurring_meetings = self.search([('recurrent_id', '=', self.id), '|', ('active', '=', False), ('active', '=', True)])
 
         for meeting in recurring_meetings:
             rset1._exdate.append(todate(meeting.recurrent_id_date))
-        return [d.astimezone(pytz.UTC) for d in rset1]
+            if both:
+                rset2._exdate.append(todate(meeting.recurrent_id_date))
+        if both:
+            return zip([d.astimezone(pytz.UTC) for d in rset1], [d.astimezone(pytz.UTC) for d in rset2])
```

Comment:
Different return type depending on method arguments is never a good idea.

You should have 2 methods (one can call the other)